src/NServiceBus.Core/Recoverability/SecondLevelRetries/DelayedRetryAction.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using DelayedDelivery;
+    using DeliveryConstraints;
+    using Extensibility;
+    using Routing;
+    using Transports;
+
+    class DelayedRetryAction
+    {
+        string timeoutManagerAddress;
+        readonly IDispatchMessages dispatcher;
+        readonly string endpointInputQueue;
+
+        public DelayedRetryAction(string endpointInputQueue, IDispatchMessages dispatcher, string timeoutManagerAddress = null)
+        {
+            this.timeoutManagerAddress = timeoutManagerAddress;
+            this.dispatcher = dispatcher;
+            this.endpointInputQueue = endpointInputQueue;
+        }
+
+        public Task Retry(IncomingMessage message, TimeSpan delay, ContextBag context)
```

Comment:
I thought that we will have `DelayedRetryExecutor` and `MoveToErrorsExecutor` and `RecoveryAction` objects would be simple data placeholders